package oce.searchcache;

message NetWorkInfo
{
  required int32 id = 1;
  optional int32 stage = 2;
  optional string name = 3;
}

message ElementarySchoolInfo
{
  required int32 id = 1;
  optional int32 year = 2;
  optional string name = 3;
}

message JuniorSchoolInfo
{
  required int32 id = 1;
  optional int32 year = 2;
  optional string name = 3;
}

message HighSchoolInfo
{
  required int32 id = 1;
  optional int32 year = 2;
  optional string name = 3;
  optional string class1 = 4;
  optional string class2 = 5;
  optional string class3 = 6;
 
}

message CollegeInfo
{
  required int32 id = 1;
  optional int32 year = 2;
  optional string name = 3;
  optional string department = 4; 
}

message UniversityInfo
{
  required int32 id = 1;
  optional int32 year = 2;
  optional string name = 3;
  optional string department = 4;
  optional string dorm = 5;
}

message WorkplaceInfo{
  required int32 id = 1;
  optional string name = 2;
}

message RegionInfo{
  required int32 id = 1;
  optional string city = 2;
  optional string province = 3;
}


message DistSearchCacheData
{
  optional string name = 1;
  optional string gender = 2;
  optional string headUrl = 3;
  optional string tinyUrl = 4;
  optional int32 status = 5;
  optional int32 state = 6;
  optional int32 stage = 7;
  optional int32 safeStatus = 8;
  optional int32 browseConfig = 9;
  optional int32 statusConfig = 10;
  optional int32 basicConfig = 11;
  optional string profilePrivacy = 12;
  repeated NetWorkInfo networks = 13;
  repeated ElementarySchoolInfo elementarySchools = 14;
  repeated JuniorSchoolInfo juniorSchools = 15;
  repeated HighSchoolInfo highSchools = 16;
  repeated CollegeInfo collegeSchools = 17;
  repeated UniversityInfo universitySchools = 18;
  repeated WorkplaceInfo workplaces = 19;
  repeated RegionInfo regions = 20;
  optional string homeCity = 21;
  optional string homeProvince = 22;
  optional string astrology = 23;
  optional int32 birthYear = 24;
  optional int32 birthMonth = 25;
  optional int32 birthDay = 26;
}


message DistSearchCachePageData
{
  optional string name = 1;
  optional string headUrl = 2;
  optional string tinyUrl = 3;
  optional int32 fansCount = 4;
  optional string subType = 5;
  optional int32 pType = 6;
}

message DistSearchCacheAppData
{
  optional string name = 1;
  optional string headUrl = 2;
  optional string tinyUrl = 3;
  optional string description = 4;
  optional string company = 5;
  optional string sideNavUrl = 6;
  optional string logo = 7;
}
